scipio/src/sys/uring.rs

```diff
@@ -490,7 +490,10 @@ macro_rules! queue_request_into_ring {
 
 macro_rules! queue_storage_io_request {
     ($self:expr, $source:ident, $op:expr) => {{
-        queue_request_into_ring!($self.poll_ring, $source, $op)
+        match $source.raw {
```

Comment:
This is good. If you take my suggestion to encode pollable in the Source (either a new `SourceType` or a boolean inside the existing `SourceType`) you can just match here by type instead of raw.

The overall direction is solid, just the specific placement (rawfd vs sourcetype) is not.